### DISCUSSION TITLE
Firefox could not download the search plugin

### DISCUSSION THREAD

**Discussion by MorsMortium** [searxng/searxng#4043] [Q&A] [answered]
Not sure if docker related or not.

I run the container with 
```
sudo docker run --rm \
             -d -p 8888:8080 \
             -v "${PWD}/searxng:/etc/searxng" \
             -e "BASE_URL=http://localhost:8888/" \
             -e "INSTANCE_NAME=my-instance" \
             searxng/searxng
```
from https://docs.searxng.org/admin/installation-docker.html
Except replacing localhost with my own IP, but tried leaving at localhost with same result.

This is the full message:
```
Firefox could not download the search plugin from: http://{IP}:8888/opensearch.xml?method=POST&autocomplete=
```
(I replaced my IP with {IP})

```
$ curl 'http://{IP}:8888/opensearch.xml'
<?xml version="1.0" encoding="utf-8"?>
<OpenSearchDescription xmlns="http://a9.com/-/spec/opensearch/1.1/" xmlns:moz="http://www.mozilla.org/2006/browser/search/">
  <ShortName>my-instance</ShortName>
  <LongName>SearXNG metasearch</LongName>
  <Description>SearXNG is a metasearch engine that respects your privacy.</Description>
  <InputEncoding>UTF-8</InputEncoding>
  <Image type="image/png">http://localhost:8888/static/themes/simple/img/favicon.png</Image>
  <Url rel="results" type="text/html" method="POST" template="http://localhost:8888/search">
    <Param name="q" value="{searchTerms}" />
  </Url>
  <Url rel="self" type="application/opensearchdescription+xml"  method="POST" template="http://localhost:8888/opensearch.xml" />
  <Query role="example" searchTerms="SearXNG" />
  <moz:SearchForm>http://localhost:8888/search</moz:SearchForm>
</OpenSearchDescription>
```

I'm not sure that it's right, that even with set IP, this file refers to localhost.

**Comment by return42** [comment]
Next time, please don't open an issue / I moved your issue to discussions.

- https://github.com/searxng/searxng-docker/discussions/188

----

Remove `-e "BASE_URL=http://localhost:8888/" \` from `docker run` command

**Comment by MorsMortium** [comment]
@return42  Thank you, I think that should be added to this page: https://docs.searxng.org/admin/installation-docker.html

**Comment by MorsMortium** [comment]
Oops. I was too fast, mixed up the add button with it being added. This didn't change it, same message

**Comment by MorsMortium** [comment]
Sorry @unixfox, but this isn't the answer, it does not work

**Reply by unixfox** [reply]
ok. you should use this: https://github.com/searxng/searxng-docker

which include everything needed to run searxng out of the box

**Reply by MorsMortium** [reply]
@unixfox Isn't that what's used by https://hub.docker.com/r/searxng/searxng? The docs mention using that, and not this repo directly, only that that image is made from the source here

**Reply by unixfox** [reply]
yes but it comes with all the needed settings out of the box.

**Reply by MorsMortium** [reply]
@unixfox Does https://hub.docker.com/r/searxng/searxng not come with those needed settings? The docs should then mention using this repo and not the image at dockerhub

**Reply by unixfox** [reply]
The docker part of the docs has been left "unmaintained" for a long time. Since there is an internal conflict around the format of the doc (rSt vs markdown).

The guy that maintain the doc doesn't use Docker. Feel free to submit a PR for improving the docker part of the doc.

**Comment by return42** [comment, marked as answer]
@MorsMortium For a lively discussion its recommended to participate from the SearXNG community at [#searxng:matrix.org](https://matrix.to/#/#searxng:matrix.org).
